text/0032-access_command.md

```diff
@@ -0,0 +1,215 @@
+  publish        list dataset for others to see and pull
+  unpublish      unlist dataset, removing it from public visibility
+  encrypt        restrict access to a dataset with a secret key
+  decrypt        remove access restrictions from a dataset
+  allow          give permissions to a peer for an encrypted dataset
+  deny           remove peer access permissions
+```
+
+### `qri access info`
+```
+$ qri access info --help
+Info describes permissions for a dataset.
+
+Usage
+  qri access info [DATASET] [DATASET...]
+```
+
+An example output for a published dataset:
+```
+$ qri access info me/dataset
+can-edit:  true
+published: true
+
+You can edit this dataset.
+
+This dataset is "published". Other users may see it when you're online, and if
+pushed to the registry it will show up in feeds for other users to browse and
+pull.
+```
+
+An example for a dataset you _don't_ own:
+```
+$ qri access info ca-state-parks/park-features
+can-edit:  false
+published: true
+
+You cannot edit this dataset, attempting to save to it's history will create a 
+fork.
+
+This dataset is "published". Other users may see it when you're online, and if
+pushed to the registry it will show up in feeds for other users to browse and
+pull.
+```
+
+`$ qri access info me/dataset` for now will only but it will be _very_ helpful for viewing dataset permsissions that control both who can access a dataset and what they can do with it.
+
+### Publish on registry push
+Pushing unencrypted datasets† to the registry _requires_ a dataset be published. To keep the old behaviour of a one-liner publish, pushing to the registry will automatically run publish on the user's behalf. The push command will present this as user feedback:
+
+```
+$ qri push me/some_dataset
+setting dataset access to "published"... done.
+pushing 28 versions of chriswhong/some_dataset...
+X of XX blocks pushed for version 12 August 2019 (Qmss4h552h3j2h33)...
+```
+
+Remotes, on the other hand _can_ accept unpublished datasets. Remote configuration will get a `requirePublish` configuration parameter to get the same registry-like behaviour. Pushing unpublished datasets is aimed at the collaborator use-case, where a datasets are explicitly pushed to trusted colleagues. when a user is pushed an unpublished dataset, it will show up when they run `qri list` locally, but this dataset won't be broadcast back to the network.
+
+† _encrypted datasets don't exist yet._
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### The `Published` field
+For some time we've modelled `published` as a boolean field outside the dataset itself, stored as a logbook operation. We've struggled with this field not capturing the problem of _where_ a dataset's been published. Re-conceiving of `published` as a "permission bit" clarifies the intent of this field. Published now answers "can the existence of this dataset be broadcasted" for the _entire history_ of a dataset. 
```

Comment:
> We've struggled with this field not capturing the problem of _where_ a dataset's been published.

I [asked some questions around this](https://github.com/qri-io/rfcs/pull/49#discussion_r410790390) over on the push/pull PR, but I’m curious about the *where* part here, and am not sure how this proposal addresses that (assuming I understand future directions).

Specifically, there’s been talk in the past about a future where there are many registries, and of course we have the situation where you could push to many peers today. If `published` is going to be modeled as a “permission bit,” is it doing so on a particular (peer/remote/registry + dataset) combination, or just on the dataset? Does marking it as published mark it as published *everywhere*? If not, it’s unclear in here how how mark it as published in all the various locations it might be. (I’m also unclear as to whether qri tracks the various peers/remotes a dataset might have been pushed to, other than the one registry you have configured for all datasets in your repo.) Or are we always just talking about the registry your have currently configured?

Reply:
> If published is going to be modeled as a “permission bit,” is it doing so on a particular (peer/remote/registry + dataset) combination, or just on the dataset?

Just on the dataset. Marking a dataset as published (we're now calling it "visible") is the same anywhere a dataset is pushed. If you don't want a user to have a hidden dataset, don't push it to them.

> I’m also unclear as to whether qri tracks the various peers/remotes a dataset might have been pushed to, other than the one registry you have configured for all datasets in your repo

Yep! We track this by adding a "push" operation to the dataset oplog that also stores the multiaddress that was pushed to. This is the same for remotes or the registry (wherever possible we model the registry as just another remote). 

Upside of this: you can get hints about where a dataset is replicated by looking at logbook data.

There's work to do here to _optimistically add the operation to the pushed logbook_, and rollback on failure but it's a technical detail.